--- a/config/environment.rb	2011-05-29 14:12:12.000000000 +0200
+++ b/config/environment.rb	2011-05-29 14:38:39.000000000 +0200
@@ -17,8 +17,8 @@
   # Specify gems that this application depends on and have them installed with rake gems:install
   config.gem "safemode"
   config.gem "rack", :version => '1.1.0'
-  config.gem "rest-client", :version => '>=1.4.0', :lib => "restclient"
-  config.gem "json"
+  # config.gem "rest-client", :version => '>=1.4.0', :lib => "restclient"
+  # config.gem "json"
   # config.gem "bj"
   # config.gem "hpricot", :version => '0.6', :source => "http://code.whytheluckystiff.net"
   # config.gem "sqlite3-ruby", :lib => "sqlite3"
